learn-arai/app/hooks/useUser.ts

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import { useLocalStorage } from './useLocalStorage';
+
+export type User = {
+    email: string;
+    password: string;
+    expires_at: string;
+};
+
+export const useUser = () => {
+    const [user, setUser] = useState<User | null>(null);
+    const { setItem, removeItem, getItem } = useLocalStorage();
+
+    useEffect(() => {
+        const userFromLocalStorage = getItem('user');
+        const isUserFromLocalStorageExist = userFromLocalStorage;
```

Comment:
Why creating duplication variable?????

Reply:
I thought, it was meaningful than just `!userFromLocalStorage`